index.html

```diff
@@ -1851,39 +1851,279 @@ <h5><code>invokeaction</code></h5>
+              </li>
+            </ul>
+          </p>
+          <pre class="example">
+          GET /things/lamp/events/overheated HTTP/1.1
+          Host: mythingserver.com
+          Accept: text/event-stream
+          Connection: keep-alive
+          </pre>
+          <p class="note" title="Opening a connection using JavaScript">
+            A Server-Sent Events connection can be initiated in JavaScript 
+            [[ECMASCRIPT]] using an <code>EventSource</code> 
+            [[EVENTSOURCE]] object.
+            <pre class="example">
+              const overheatedEventSource = new EventSource('/things/lamp/events/overheated');
+            </pre>
+          </p>
+          <p class="rfc2119-assertion" 
+          id="core-profile-protocol-binding-events-subscribeevent-3">
+            If a Web Thing receives an HTTP request following the format
+            above and the Consumer has permission to subscribe to the 
+            corresponding event, then it MUST follow the Server-Sent Events 
+            [[EVENTSOURCE]] specification to maintain an open connection with 
+            the Consumer and push event data to the Consumer as events of the 
+            specified type are emitted.
           </p>
           <p>
-            <code>subscribeevent</code>, <code>unsubscribeevent</code>,
-            <code>subscribeallevents</code> and
-            <code>unsubscribeallevents</code> would require consensus on a
-            default event subscription mechanism for HTTP (e.g. Server Sent
-            Events or WebSockets).
+            This will usually involve the Web Thing initially sending an HTTP 
+            response to the Consumer with:
+            <ul>
+              <li>Status code set to <code>200</code></li>
+              <li>
+                <code>Content-Type</code> header set to 
+                <code>text/event-stream</code>
+              </li>
+            </ul>
+          </p>
+          <pre class="example">
+          HTTP/1.1 200 OK
+          Content-Type: text/event-stream
+          </pre>
+          <p class="rfc2119-assertion" 
+            id="core-profile-protocol-binding-events-subscribeevent-4">
+            Whenever an event of the specified type occurs while the Web Thing 
+            has an open connection with a Consumer, the Web Thing MUST 
+            send event data to the Consumer using the event stream format in the 
+            Server-Sent Events [[EVENTSOURCE]] specification. For each message 
+            sent, the Web Thing MUST set the <code>event</code> field to the 
+            name of the <code>EventAffordance</code> and populate the 
+            <code>data</code> field with event data, if any. The 
+            event data MUST follow the data schema specified in the
+            <code>EventAffordance</code> and MUST be serialized in JSON.
+            The <code>id</code> field SHOULD be set to a unique identifier for 
```

Comment:
should this identifier be a URL?  Should we treat subscriptions as resources, so that a consumer can reconnect to a subscription if a connection is dropped?  In general, I need to look at what SSE says about this... and we should follow it, as far as we can

Reply:
SSE has a built-in mechanism for resuming dropped connections, it's not necessary to manually keep track of open connections to do this. As far as I can tell IDs can be anything (e.g. an integer or a timestamp), but I don't think it's common to use a URL.